src/detection/cpu/cpu_linux.c

```diff
@@ -197,6 +197,86 @@ const char* ffDetectCPUImpl(const FFCPUOptions* options, FFCPUResult* cpu)
                 }
             }
 
+            // In Asahi Linux, reading /proc/device-tree/compatible gives
+            // information on the device model. It consists of 3 NUL terminated
+            // strings, the second of which gives the actual SoC model. But it
+            // is not the marketing name, i.e. for M2 there is "apple,t8112" in
+            // the compatible string.
+            //
+            // A full list of the SoC model names can be found here:
+            // https://github.com/AsahiLinux/docs/wiki/Codenames
+            #define DT_COMPAT_PATH "/proc/device-tree/compatible"
```

Comment:
Maybe we should guard this with `#ifdef __aarch64__`

Reply:
Sure.